scheduler/src/cook/mesos/rebalancer.clj

```diff
@@ -382,12 +382,28 @@
             :to-make-room-for pending-job)]
     [state nil]))
 
+(defn reserve-hosts!
+  "Reserves all hosts in preemption-decisions which will preempt more than one task"
+  [rebalancer-reservation-atom preemption-decisions]
+  (let [multiple-task-decisions (filter #(< 1 (count (:task %))) preemption-decisions)
+        reservations (->> multiple-task-decisions
+                          (map (fn [d] [(:job/uuid (:to-make-room-for d))
+                                        (:hostname d)]))
+                          (into {}))]
+    (swap! rebalancer-reservation-atom (fn [{:keys [launched-job-uuids]}]
+                                        ; In case one of the jobs the rebalancer has launched already while computing
+                                        ; the pre-emption decisions, remove it's reservation from the map. Then
```

Comment:
I don't understand the grammar here:

> In case one of the jobs the rebalancer has launched already while computing the pre-emption decisions, remove it's reservation from the map

Reply:
Me neither. Fixed.